hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/ContainerHealthSchemaManager.java

```diff
@@ -69,8 +71,14 @@ public ContainerHealthSchemaManager(
     SelectQuery<Record> query = dslContext.selectQuery();
     query.addFrom(UNHEALTHY_CONTAINERS);
     if (state != null) {
-      query.addConditions(
-          UNHEALTHY_CONTAINERS.CONTAINER_STATE.eq(state.toString()));
+      if (state.equals(ALL_REPLICAS_UNHEALTHY)) {
+        query.addConditions(UNHEALTHY_CONTAINERS.CONTAINER_STATE
+            .eq(UNDER_REPLICATED.toString()));
+        query.addConditions(UNHEALTHY_CONTAINERS.ACTUAL_REPLICA_COUNT.eq(0));
```

Comment:
Q: In which other cases are `ACTUAL_REPLICA_COUNT` set? Apart from this `ALL_REPLICAS_UNHEALTHY` case.